src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexWriter.cs

```diff
@@ -42,6 +47,10 @@ public static RegexCode Write(RegexTree tree)
         {
             var writer = new RegexWriter(stackalloc int[EmittedSize], stackalloc int[IntStackSize]);
             RegexCode code = writer.RegexCodeFromRegexTree(tree);
+            if (writer.ahoCorasickWords != null)
```

Comment:
no need to check?

Reply:
in fact, can't `RegexCodeFromRegexTree()` set this in code

Reply:
Yup, fixed.